moztelemetry/dataset.py

```diff
@@ -176,6 +176,9 @@ def records(self, sc, limit=None, sample=1, decode=None, summaries=None):
         if summaries and limit is None and sample != 1:
             if sample < 0 or sample > 1:
                 raise ValueError('sample must be between 0 and 1')
+            # We want this sample to be reproducible.
+            # See https://bugzilla.mozilla.org/show_bug.cgi?id=1318681
+            random.seed(42)
```

Comment:
I added a `seed` parameter to cover this use case as well as the case where we want to preserve the current behavior (no seed)